copybookreader-plugins/src/main/java/co/cask/hydrator/plugin/batch/CopybookSource.java

```diff
@@ -147,9 +134,17 @@ private Object parseValue(Schema.Field field, AbstractFieldValue value) throws I
         return value.asFloat();
       case STRING:
         return value.asString();
+      case UNION:
+        if (field.getSchema().isNullableSimple()) {
+          try {
+            return parseValue(Schema.Field.of(field.getName(), field.getSchema().getNonNullable()), value);
+          } catch (Exception e) {
```

Comment:
rather than relying on the recursive call throwing a null pointer exception and catching it, it is cleaner to just check if the value is null before trying to parse it. If so, return it.